Bio/PDB/cealign.py

```diff
@@ -110,36 +110,70 @@ def align(self, structure, transform=True):
             raise PDBException(msg)
 
         # Run CEAlign
-        # CEAlign returns the best N paths, where each path is a pair of lists
-        # with aligned atom indices.
+        # CEAlign returns the best N paths, sorted descending by length,
+        # where each path is a pair of lists with aligned atom indices.
         paths = run_cealign(self.refcoord, coord, self.window_size, self.max_gap)
-        longest_length = len(paths[0][0])
         longest_paths = [
-            (tuple(pA), tuple(pB)) for pA, pB in paths if len(pA) == longest_length
+            path for path in paths if len(path[0][0]) == len(paths[0][0][0])
         ]
 
         # Iterate over paths and find the one that gives the lowest
         # corresponding RMSD. Use QCP to align the molecules.
-        best_rmsd, best_u = 1e6, None
-        for path in longest_paths:
+        aln = QCPSuperimposer()
+        best_rmsd = float("inf")
+        best_rigid_motion = None
+        best_path = None
+        best_z_score = None
+        for path, z_score in longest_paths:
             idxA, idxB = path
 
             coordsA = np.array([self.refcoord[i] for i in idxA])
             coordsB = np.array([coord[i] for i in idxB])
 
-            aln = QCPSuperimposer()
             aln.set(coordsA, coordsB)
             aln.run()
             if aln.rms < best_rmsd:
                 best_rmsd = aln.rms
-                best_u = (aln.rot, aln.tran)
+                best_rigid_motion = (aln.rot, aln.tran)
+                best_path = path
+                best_z_score = z_score
 
-        if best_u is None:
+        if best_rigid_motion is None:
             raise RuntimeError("Failed to find a suitable alignment.")
 
+        # Gap optimization
```

Comment:
I'd move this to a function of its own and add a kwarg that enables/disables this optimization from the top-level cealign call. 

Also, I'm surprised this is still so fast done in Python vs done in C!